pkg/handler/cli/article_handler.go

```diff
@@ -0,0 +1,207 @@
+			if err != nil {
+				log.Fatalf("Error fetching articles by source: %v", err)
+			}
+			filteredArticles = append(filteredArticles, sourceArticles...)
+		}
+	} else {
+		filteredArticles, err = h.Service.GetAll()
+		if err != nil {
+			errors.New("error fetching all articles")
+		}
+	}
+
+	if keywords != "" {
+		keywordList := strings.Split(keywords, ",")
+		var keywordFilteredArticles []model.Article
+		for _, keyword := range keywordList {
+			keywordArticles, err := h.Service.GetByKeyword(strings.TrimSpace(keyword))
+			if err != nil {
+				log.Fatalf("Error fetching articles by keyword: %v", err)
+			}
+			keywordFilteredArticles = append(keywordFilteredArticles, keywordArticles...)
+		}
+		filteredArticles = intersect(filteredArticles, keywordFilteredArticles)
+	}
+
+	if dateStart != "" || dateEnd != "" {
+		dateRangeArticles, err := h.Service.GetByDateInRange(dateStart, dateEnd)
+		if err != nil {
+			log.Fatalf("Error fetching articles by date range: %v", err)
+		}
+		filteredArticles = intersect(filteredArticles, dateRangeArticles)
+	}
+
+	uniqueArticles := make(map[int]model.Article)
+	for _, article := range filteredArticles {
+		uniqueArticles[article.Id] = article
+	}
+
+	var result []model.Article
+	for _, article := range uniqueArticles {
+		result = append(result, article)
+	}
+	return result
+}
+
+// printArticles prints the provided articles to the console using template.
+func (h *Handler) printArticles(articles []model.Article, sources, keywords, dateStart, dateEnd string) {
+	tmplPath := getTemplatePath()
+	tmpl, err := template.New("article.tmpl").Funcs(sprig.FuncMap()).Funcs(template.FuncMap{
+		"groupBy": groupBy,
+		"gt1": func(s string) bool {
+			return len(s) > 1
+		},
+		"highlightKeywords": func(text string) string {
+			if keywords == "" {
+				return text
+			}
+			return highlightKeywords(text, keywords)
+		},
+	}).ParseFiles(tmplPath)
+	if err != nil {
+		log.Fatalf("Error parsing template: %v", err)
+	}
+
+	data := struct {
+		Articles  []model.Article
+		Sources   string
+		Keywords  string
+		DateStart string
+		DateEnd   string
+	}{
+		Articles:  articles,
+		Sources:   sources,
+		Keywords:  keywords,
+		DateStart: dateStart,
+		DateEnd:   dateEnd,
+	}
+
+	err = tmpl.ExecuteTemplate(os.Stdout, "page", data)
+	if err != nil {
+		log.Fatalf("Error executing template: %v", err)
+	}
+}
+
+// getTemplatePath returns the path to the template file.
+func getTemplatePath() string {
+	possiblePaths := []string{
+		"../../templates/article.tmpl",
+		"templates/article.tmpl",
+		"../../../templates/article.tmpl",
+	}
+
+	for _, path := range possiblePaths {
+		if _, err := os.Stat(path); err == nil {
+			return path
+		}
+	}
+
+	log.Fatalf("Template file not found in any of the expected paths")
+	return ""
+}
+
+// intersect returns the common elements between two slices of articles.
+func intersect(a, b []model.Article) (articles []model.Article) {
```

Comment:
check strings pkg, it should have the same func

Reply:
Checked. There is not a built-in solution, but I've moved it to filter package, because it's used only there.